docs/deployments/realtime-api/predictors.md

```diff
@@ -492,58 +492,206 @@ If your application requires additional dependencies, you can install additional
+
+```python
+import requests
+
+url = "https://***.amazonaws.com/my-api"
+requests.post(url, json={"key": "value"})
+```
+
+Or if you have a json string:
+
+```python
+import requests
+import json
+
+url = "https://***.amazonaws.com/my-api"
+jsonStr = json.dumps({"key": "value"})
+requests.post(url, data=jsonStr, headers={"Content-Type": "application/json"})
+```
+
+##### CURL
 
 ```bash
-$ curl http://***.amazonaws.com/my-api \
+$ curl https://***.amazonaws.com/my-api \
     -X POST -H "Content-Type: application/json" \
     -d '{"key": "value"}'
 ```
 
-The `payload` parameter type will be a `bytes` object if a request with a `Content-Type: application/octet-stream` is made:
+Or if you have a json file:
 
 ```bash
-$ curl http://***.amazonaws.com/my-api \
-    -X POST -H "Content-Type: application/octet-stream" \
-    --data-binary @file.bin
+$ curl https://***.amazonaws.com/my-api \
+    -X POST -H "Content-Type: application/json" \
+    -d @file.json
 ```
 
-The `payload` parameter type will be a `bytes` object if a request doesn't have the `Content-Type` set:
+#### Reading the payload
 
-```bash
-$ curl http://***.amazonaws.com/my-api \
-    -X POST -H "Content-Type:" \
-    -d @sample.txt
+When sending a JSON payload, the `payload` parameter will be a Python object:
+
+```python
+class PythonPredictor:
+    def __init__(self, config):
+        pass
+
+    def predict(self, payload):
+        print(payload["key"])  # prints "value"
+```
+
+### Binary data
+
+#### Making the request
+
+##### Python
+
+```python
+import requests
+import pickle
+
+url = "https://***.amazonaws.com/my-api"
+pklBytes = pickle.dumps({"key": "value"})
+requests.post(url, data=pklBytes, headers={"Content-Type": "application/octet-stream"})
 ```
 
-The `payload` parameter type will be a `starlette.datastructures.FormData` object if a request with a `Content-Type: multipart/form-data` is made:
+##### CURL
 
 ```bash
-$ curl http://***.amazonaws.com/my-api \
-    -X POST -H "Content-Type: multipart/form-data" \
-    -F "fieldName=@file.txt"
+$ curl https://***.amazonaws.com/my-api \
+    -X POST -H "Content-Type: application/octet-stream" \
+    --data-binary @object.pkl
+```
+
+#### Reading the payload
+
+Since the `Content-Type: application/octet-stream` header is used, the `payload` parameter will be a `bytes` object:
+
+```python
+import pickle
+
+class PythonPredictor:
+    def __init__(self, config):
+        pass
+
+    def predict(self, payload):
+        obj = pickle.loads(payload)
+        print(obj["key"])  # prints "value"
 ```
 
-The `payload` parameter type will be a `starlette.datastructures.FormData` object if a request with a `Content-Type: application/x-www-form-urlencoded` is made:
+Here's an example if the binary data is an image:
+
+```python
+from PIL import Image
+import io
+
+class PythonPredictor:
+    def __init__(self, config):
+        pass
+
+    def predict(self, payload, headers):
+        img = Image.open(io.BytesIO(payload))
```

Comment:
Maybe a short comment describing this would be good (i.e. `loads object.pkl as an image`).